libraries/mbed/targets/cmsis/TARGET_WIZNET/TARGET_W7500x/W7500x_i2c.c

```diff
@@ -97,27 +104,40 @@ void I2C_SendData(I2C_TypeDef* I2Cx,uint16_t Data)
 
 int8_t I2C_SendDataAck(I2C_TypeDef* I2Cx,uint16_t Data)
 {
-    I2Cx -> TXR = (uint16_t)Data;
-    if(I2C_CheckEvent(I2Cx,I2C_ACKR) == ERROR)
+   buf[0] = Data;
+    if(buf[0] == buf[1])
```

Comment:
This one is an entry point for sending data. What's the intention here with internal buffer (buf)?